### PR TITLE
Fix some pdf related issues

### LINKED ISSUE #108: Some documents leads to an issue
```
2018-10-11 14:09:25 [scrapy.core.scraper] ERROR: Error processing {'title': 'Review of Historic Royal Palaces', 'uri': 'https://www.hrp.org.uk/media/1129/sustainabilityreport2013-14.pdf', 'provider': None}
Traceback (most recent call last):
File "/usr/local/lib/python3.6/site-packages/twisted/internet/defer.py", line 654, in _runCallbacks
current.result = callback(current.result, *args, **kw)
File "/wsf_scraper/wsf_scraping/pipelines.py", line 131, in process_item
full_item = self.check_keywords(item, spider.name, item['pdf'])
File "/wsf_scraper/wsf_scraping/pipelines.py", line 75, in check_keywords
pdf_file, pdf_text = parse_pdf_document(f)
File "/wsf_scraper/pdf_parser/pdf_parse.py", line 40, in parse_pdf_document
html_file = open(parsed_path, 'rb')
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/tmp1sd6ok4x/sustainabilityreport2013-14.xml'
```

Probably coming from a non-convertible document. 
We are catching errors coming from pdftohtml, so it may come from somewhere else?

### LINKED ISSUE #109: Parliament spider sometimes fail to recognize PDFs
```
2018-10-11 14:11:59 [scrapy.core.scraper] ERROR: Spider error processing <GET https://www.gmmh.nhs.uk/download.cfm?doc=docm93jijm4n3073.pdf&ver=4810> (referer: http://search-material.parliament.uk/search?Parameters.Fields.all-target=&Parameters.Fields.all=&Parameters.Fields.any-target=&Parameters.Fields.any=&Parameters.Fields.date=01%2F01%2F1980&Parameters.Fields.date=04%2F10%2F2018&Parameters.Fields.exclude-target=&Parameters.Fields.exclude=&Parameters.Fields.house=House+of+Commons&Parameters.Fields.house=House+of+Lords&Parameters.Fields.member=&Parameters.Fields.phrase-target=&Parameters.Fields.phrase=&Parameters.Fields.reference=&Parameters.Fields.subject=&Parameters.PageSize=100&When%3A=date&Parameters.Page=13)
Traceback (most recent call last):
File "/usr/local/lib/python3.6/site-packages/scrapy/utils/defer.py", line 102, in iter_errback
yield next(it)
File "/usr/local/lib/python3.6/site-packages/scrapy/spidermiddlewares/offsite.py", line 30, in process_spider_output
for x in result:
File "/usr/local/lib/python3.6/site-packages/scrapy/spidermiddlewares/referer.py", line 339, in <genexpr>
return (_set_referer(r) for r in result or ())
File "/usr/local/lib/python3.6/site-packages/scrapy/spidermiddlewares/urllength.py", line 37, in <genexpr>
return (r for r in result or () if _filter(r))
File "/usr/local/lib/python3.6/site-packages/scrapy/spidermiddlewares/depth.py", line 58, in <genexpr>
return (r for r in result or () if _filter(r))
File "/wsf_scraper/wsf_scraping/spiders/parliament_spider.py", line 98, in parse_others
for href in response.css('a::attr(href)').extract():
File "/usr/local/lib/python3.6/site-packages/scrapy/http/response/__init__.py", line 99, in css
raise NotSupported("Response content isn't text")
scrapy.exceptions.NotSupported: Response content isn't text
```

Probably coming from urls not ending with `.pdf`. To investigate deeper

### PR DESCRIPTION
**Pull request by SamDepardieu**
# Description

This PR aims to fix a few PDF related issues:
- Not converted PDFs leading to an error (fix #108) 
- Parliament spider not recognizing some pages as pdf (fix #109)

## Type of change

Please delete options that are not relevant.

- [x] :bug: Bug fix (Add `Fix #(issue)` to your PR)

# How Has This Been Tested?

Local tests
# Checklist:

- [x] My code follows the style guidelines of this project (pep8 AND pyflakes)
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] If needed, I changed related parts of the documentation
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] If my PR aims to fix an issue, I referenced it using `#(issue)`

### CODE REVIEW

File: pdf_parser/pdf_parse.py [hunk 1]
```diff
@@ -20,25 +20,26 @@ def parse_pdf_document(document):
...
             document.name,
             e.stderr,
         )
-        return None
-
-    html_file = open(parsed_path, 'rb')
-    soup = bs(html_file.read(), 'html.parser')
+        return None, None
+    try:
+        html_file = open(parsed_path, 'rb')
```
Comment (hblanks) [inline pdf_parser/pdf_parse.py]: Do we close this file somewhere, or would a context manager be more appropriate here?

Comment (SamDepardieu) [reply pdf_parser/pdf_parse.py]: It's closed at the end of the function. It could be in a context manager if you think it's better

Comment (hblanks) [reply pdf_parser/pdf_parse.py]: Usually best to close your file descriptions as soon as you're done with them -- if you had an exception between here and end of the function, you'd end up with an fd leak. But, up to you whether to fix now or later.

**Comment by SamDepardieu** [comment]
@hblanks I pushed a new version with a context manager 👌

**Pull request merged**